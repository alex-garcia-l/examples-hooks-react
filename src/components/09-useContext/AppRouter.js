import React from 'react';
import {
  Link,
  useLocation,
  Outlet
} from "react-router-dom";

export const AppRouter = () => {

  let { pathname } = useLocation();

  console.log(pathname);

  return (
    <div>
      <Outlet />
    </div>
  );
};
