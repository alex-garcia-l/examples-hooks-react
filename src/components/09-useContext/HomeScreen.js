import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomeScreen = () => {

  const { user } = useContext(UserContext);

  return (
    <div className="p-5 mb-4 bg-light rounded-3">
      <div className="container-fluid py-5">
        <h2 className="display-5 fw-bold text-center">HomeScreen</h2>
        <hr />
        <pre className="col-md-8 fs-4">{JSON.stringify(user, null, 3)}</pre>
      </div>
    </div>
  );
};