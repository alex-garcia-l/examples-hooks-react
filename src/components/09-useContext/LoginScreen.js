import React, { useContext } from 'react';
import Swal from 'sweetalert2';

import { UserContext } from './UserContext';

const Toast = Swal.mixin({
  toast: true,
  position: 'top-end',
  showConfirmButton: false,
  timer: 3000,
  timerProgressBar: true,
  didOpen: (toast) => {
    toast.addEventListener('mouseenter', Swal.stopTimer)
    toast.addEventListener('mouseleave', Swal.resumeTimer)
  }
});

export const LoginScreen = () => {

  const { setUser } = useContext(UserContext);

  const handleClick = (evt) => {
    evt.preventDefault();

    setUser({
      id: 1234,
      name: 'Alex',
      user: '@alex'
    });

    Toast.fire({
      icon: 'success',
      title: 'Login success!'
    });
  }

  return (
    <div className="p-5 mb-4 bg-light rounded-3">
      <div className="container-fluid py-5">
        <h2 className="display-5 fw-bold text-center">LoginScreen</h2>
        <hr />
        <div className="text-center">
          <button
            className="btn btn-success btn-lg"
            onClick={ handleClick }
          >
            Login
          </button>
        </div>
      </div>
    </div>
  );
};