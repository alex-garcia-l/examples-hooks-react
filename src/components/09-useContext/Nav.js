import React from 'react';
import { NavLink } from 'react-router-dom';

export const Nav = () => {
  return (

    <div className="container mb-5">
      <header className="d-flex justify-content-center py-3">
        <ul className="nav nav-pills">
          <li className="nav-item"><NavLink className="nav-link" to="/context-app" end>Home</NavLink></li>
          <li className="nav-item"><NavLink className="nav-link" to="/context-app/about">About</NavLink></li>
          <li className="nav-item"><NavLink className="nav-link" to="/context-app/login">Login</NavLink></li>
        </ul>
      </header>
    </div>
  );
};
