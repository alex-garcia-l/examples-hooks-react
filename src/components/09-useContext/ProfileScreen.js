import React, { useContext } from 'react';
import Swal from 'sweetalert2';

import { UserContext } from './UserContext';

const Toast = Swal.mixin({
  toast: true,
  position: 'top-end',
  showConfirmButton: false,
  timer: 3000,
  timerProgressBar: true,
  didOpen: (toast) => {
    toast.addEventListener('mouseenter', Swal.stopTimer)
    toast.addEventListener('mouseleave', Swal.resumeTimer)
  }
});

export const ProfileScreen = () => {

  const { user, setUser } = useContext(UserContext);

  const handleClick = (evt) => {
    evt.preventDefault();

    setUser({});

    Toast.fire({
      icon: 'success',
      title: 'Logout success!'
    });
  }

  return (
    <div className="p-5 mb-4 bg-light rounded-3">
      <div className="container-fluid py-5">
        <h2 className="display-5 fw-bold text-center">ProfileScreen</h2>
        <hr />
        <div className="text-center">
          <button
            className="btn btn-danger btn-lg"
            onClick={ handleClick }
          >
            Logout
          </button>
        </div>
        <pre className="col-md-8 fs-4">{JSON.stringify(user, null, 3)}</pre>
      </div>
    </div>
  );
};
