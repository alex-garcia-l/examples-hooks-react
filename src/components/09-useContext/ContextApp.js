import React, { useState } from 'react';
import { Outlet } from 'react-router-dom';

import { UserContext } from './UserContext';
import { Nav } from './Nav';

export const ContextApp = () => {

  const [user, setUser] = useState({});

  return (
    <UserContext.Provider value={{ user, setUser }}>
      <h1>Context App</h1>
      <hr />
      <Nav />

      <Outlet />
    </UserContext.Provider>
  );
};
