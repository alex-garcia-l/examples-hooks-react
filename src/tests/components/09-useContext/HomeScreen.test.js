import { mount } from 'enzyme';

import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Prueba para el componente <HomeScreen />', () => {

  const user = {

  };

  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  );

  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  

});
