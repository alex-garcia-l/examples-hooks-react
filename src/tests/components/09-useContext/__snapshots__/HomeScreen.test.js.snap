// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba para el componente <HomeScreen /> debe mostrarse correctamente 1`] = `
<div
  className="p-5 mb-4 bg-light rounded-3"
>
  <div
    className="container-fluid py-5"
  >
    <h2
      className="display-5 fw-bold text-center"
    >
      HomeScreen
    </h2>
    <hr />
    <pre
      className="col-md-8 fs-4"
    >
      {}
    </pre>
  </div>
</div>
`;
